server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2304,6 +2416,14 @@ protected final void closeNoLock(String reason, CountDownLatch closedLatch) {
         }
     }
 
+    private boolean isWriteLockHeld() {
+        if (isContextAwareEnabled) {
+            return rwl.isWriteLockedByCurrentThread() && ((CompositeIndexWriter) documentIndexWriter).isWriteLockedByCurrentThread();
```

Comment:
Under what situations, would `rwl` be held by current thread, but CompositeIndexWriter'ls rwl won't be?